partitionmanager/tools.py

```diff
@@ -0,0 +1,26 @@
+"""
+Tools for working with iterators. Helpers.
+"""
+
+from itertools import tee
+
+
+def pairwise(iterable):
+    """
+    iterable -> (s0,s1), (s1,s2), (s2, s3), ... (s_n-1, s_n).
+    """
+    a, b = tee(iterable)
+    next(b, None)
+    return zip(a, b)
+
+
+def iter_show_end(iterable):
+    """
+    iterable -> (s0, false), (s1, false), ... (s_n, true).
+    """
+    it = iter(iterable)
+    last = next(it)
```

Comment:
nit: in a function which is specifically doing something special with the last element of the iterator, calling an intermediate variable `last` is confusing. `prev`, maybe?

Reply:
SGTM.